sys/include/net/ng_rpl.h

```diff
@@ -0,0 +1,423 @@
+#define NG_RPL_DEFAULT_MIN_HOP_RANK_INCREASE 256
+
+/**
+ * @brief   Maximum rank increase
+ */
+#define NG_RPL_DEFAULT_MAX_RANK_INCREASE 0
+
+/**
+ * @brief   Number of implemented Objective Functions
+ */
+#define NG_RPL_IMPLEMENTED_OFS_NUMOF 1
+
+/**
+ * @brief   Default Objective Code Point (OF0)
+ */
+#define NG_RPL_DEFAULT_OCP 0
+
+/**
+ * @name RPL Mode of Operations
+ * @{
+ */
+#define NG_RPL_MOP_NO_DOWNWARD_ROUTES  0x00
+#define NG_RPL_MOP_NON_STORING_MODE    0x01
+#define NG_RPL_MOP_STORING_MODE_NO_MC  0x02
+#define NG_RPL_MOP_STORING_MODE_MC     0x03
+/** default MOP set on compile time */
+#ifndef NG_RPL_DEFAULT_MOP
+#   define NG_RPL_DEFAULT_MOP NG_RPL_MOP_STORING_MODE_NO_MC
+#endif
+/** @} */
+
+/**
+ * @name Sequence counter handling
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-7.2">
+ *          Sequence Counter Operation
+ *      </a>
+ * @{
+ */
+#define NG_RPL_COUNTER_MAX                 255
+#define NG_RPL_COUNTER_LOWER_REGION        127
+#define NG_RPL_COUNTER_SEQ_WINDOW          16
+#define NG_RPL_COUNTER_INIT                (NG_RPL_COUNTER_MAX - NG_RPL_COUNTER_SEQ_WINDOW + 1)
+
+static inline uint8_t NG_RPL_COUNTER_INCREMENT(uint8_t counter)
+{
+    return (counter > NG_RPL_COUNTER_LOWER_REGION ?
+            (counter == NG_RPL_COUNTER_MAX ? counter = 0 : ++counter) :
+            (counter == NG_RPL_COUNTER_LOWER_REGION ? counter = 0 : ++counter));
+}
+
+static inline bool NG_RPL_COUNTER_IS_INIT(uint8_t counter)
+{
+    return (counter > NG_RPL_COUNTER_LOWER_REGION);
+}
+
+static inline bool NG_RPL_COUNTER_GREATER_THAN_LOCAL(uint8_t A, uint8_t B)
+{
+    return (((A < B) && (NG_RPL_COUNTER_LOWER_REGION + 1 - B + A < NG_RPL_COUNTER_SEQ_WINDOW))
+            || ((A > B) && (A - B < NG_RPL_COUNTER_SEQ_WINDOW)));
+}
+
+static inline bool NG_RPL_COUNTER_GREATER_THAN(uint8_t A, uint8_t B)
+{
+    return ((A > NG_RPL_COUNTER_LOWER_REGION) ? ((B > NG_RPL_COUNTER_LOWER_REGION) ?
+                NG_RPL_COUNTER_GREATER_THAN_LOCAL(A, B) : 0) :
+            ((B > NG_RPL_COUNTER_LOWER_REGION) ? 1 : NG_RPL_COUNTER_GREATER_THAN_LOCAL(A, B)));
+}
+/** @} */
+
+/**
+ * @name Trickle parameters
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-8.3.1">
+ *          Trickle Parameters
+ *      </a>
+ * @{
+ */
+#define NG_RPL_DEFAULT_DIO_INTERVAL_DOUBLINGS 20
+#define NG_RPL_DEFAULT_DIO_INTERVAL_MIN 3
+#define NG_RPL_DEFAULT_DIO_REDUNDANCY_CONSTANT 10
+/** @} */
+
+/**
+ * @name Default parent and route entry lifetime
+ * default lifetime will be multiplied by the lifetime unit to obtain the resulting lifetime
+ * @{
+ */
+#define NG_RPL_DEFAULT_LIFETIME 60
+#define NG_RPL_LIFETIME_UNIT 2
+/** @} */
+
+/**
+ * @brief Interval of the void _update_lifetime() function
+ */
+#define NG_RPL_LIFETIME_STEP 2
+
+/**
+ * @brief A DODAG can be grounded or floating
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-3.2.4">
+ *          Grounded and Floating DODAGs
+ *      </a>
+ */
+#define NG_RPL_GROUNDED 1
```

Comment:
Yes, "our" dodags are grounded by default. we do not support floating dodags at the moment